get_winner.py

```diff
@@ -0,0 +1,107 @@
+"""Takes in a dictionary of votes, and returns the meme with the highest wins to contests."""
+
+from vote.vote import MemeVote
+from collections import defaultdict
+
+
+def group_by(iterable, key):
+    """Place each item in an iterable into a bucket based on calling the key
+    function on the item.
+    """
+    group_to_items = {}
+    for item in iterable:
+        group = key(item)
+        if group not in group_to_items:
+            group_to_items[group] = []
+        group_to_items[group].append(item)
+    return group_to_items
+
+
+def get_winner_meme(vote):
+    """Simple key for groupby function in get_meme_to_wins function.
+    """
+    return vote['winner']
+
+
+def get_memes_to_participated_votes(contests):
+    """Takes in all the votes, and process them into two dictionaries, then updates the first dictionary by the second.
+    """
+    meme_in_contest_as_meme_1 = group_by(contests, get_contest_meme_1)
+    meme_in_contest_as_meme_2 = group_by(contests, get_contest_meme_2)
+    meme_in_contest_as_meme_1.update(meme_in_contest_as_meme_2)
+    return meme_in_contest_as_meme_1
```

Comment:
This will not union the keys.

```py
one = {'a': [1, 2, 3]}
two = {'a': [4, 5]}
one.update(two)
one  #> {'a': [4, 5]}
```

So this only keeps half the votes for a meme.